ports/espressif/supervisor/usb_serial_jtag.c

```diff
@@ -101,21 +101,28 @@ bool usb_serial_jtag_connected(void) {
 }
 
 char usb_serial_jtag_read_char(void) {
-    if (ringbuf_num_filled(&ringbuf) == 0) {
+    if (ringbuf_num_filled(&ringbuf) == 0 && !usb_serial_jtag_ll_rxfifo_data_available()) {
         return -1;
     }
-    char c = ringbuf_get(&ringbuf);
+    char c = -1;
+    if (ringbuf_num_filled(&ringbuf) > 0) {
+        c = ringbuf_get(&ringbuf);
+    }
     // Maybe re-enable the recv interrupt if we've emptied the ringbuf.
     if (ringbuf_num_filled(&ringbuf) == 0) {
         usb_serial_jtag_ll_disable_intr_mask(USB_SERIAL_JTAG_INTR_SERIAL_OUT_RECV_PKT);
```

Comment:
Won't this affect gdb? I should probably test it.

Reply:
I don't think so since it just disables the serial out interrupt.